### ISSUE TITLE
repo upload opens nano, breaks terminal input

### ISSUE THREAD

**Issue by jeremysalwen** [xonsh/xonsh#2355]
When I run

https://code.google.com/p/git-repo/

    $ repo upload .

from bash, it opens up my editor (nano) and allows me to edit the contents to specify which branches to upload (this is very similar to the way git opens your editor for you to enter your commit message).

However, from xonsh, when I run the same command, it opens nano, but the terminal input is broken.  Pressing the arrow keys just prints "^[OB^[OB^[OB^[OD^[OA^[OC",  page up and page down just print "^[[5~^[[6~", etc.  Return is messed up as well, and will actually cause the nano "window" to scroll up out of the terminal.

I don't know what the specific name is for what's going on, but it seems related to escaping/unescaping  control characters.    I am available to run diagnostics and debug how it got into this state.

**Comment by gforsyth** [comment]
Hey @jeremysalwen -- thanks for reporting! Could you please post the output of `xonfig`?

And does the issue persist if you first run:

```console
__xonsh_commands_cache__.threadable_predictors['repo'] = lambda args: False
```
??

**Comment by jeremysalwen** [comment]
output of xonfig.

```
+------------------+---------------------------------------------------+
| xonsh            | 0.5.9                                             |
| Git SHA          | 2ae5218                                           |
| Commit Date      | Mar 15 12:06:26 2017                              |
| Python           | 3.4.3                                             |
| PLY              | 3.9                                               |
| have readline    | True                                              |
| prompt toolkit   | 1.0.13                                            |
| shell type       | prompt_toolkit                                    |
| pygments         | None                                              |
| on posix         | True                                              |
| on linux         | <xonsh.lazyasd.LazyBool object at 0x7f64e49ffac8> |
| distro           | Ubuntu                                            |
| on darwin        | False                                             |
| on windows       | False                                             |
| on cygwin        | False                                             |
| is superuser     | False                                             |
| default encoding | utf-8                                             |
| xonsh encoding   | utf-8                                             |
| encoding errors  | surrogateescape                                   |
+------------------+---------------------------------------------------+
```

And yes, that command fixes it.

**Comment by gforsyth** [comment]
Fixed in #2357
